#!/usr/bin/env python
PACKAGE = "stdr_velodyne"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("spin_fraction", double_t, 0, "Aggregates scans until that much of a spin has been constructed.", 1, 0, 1)
gen.add("encoder_start", int_t, 0, "The encoder value for the start of a whole spin. Only used if spin_fraction is 1.", 18000, 0, 36000)
gen.add("keep_ratio", double_t, 0, "Keeps only a fraction of the constructed scan (downsampling).", 1, 0, 1)
gen.add("target_frame", str_t, 0, "The target frame. If empty, then do not transform.", "")

exit(gen.generate(PACKAGE, "stdr_velodyne", "Aggregator"))
